newdle/client/src/components/Summary.js

```diff
@@ -0,0 +1,80 @@
+import React, {useEffect, useState} from 'react';
+import {serializeDate, toMoment} from '../util/date';
+import {useDispatch, useSelector} from 'react-redux';
+import {Button, Container, Icon, Header} from 'semantic-ui-react';
+import ParticipantTable from './ParticipantTable';
+import {clearFinalDate} from '../actions';
+import {getNewdle, getFinalDate} from '../selectors';
+import {updateNewdle} from '../actions';
+import client from '../client';
+import styles from './Summary.module.scss';
+
+export default function Summary() {
+  const [submitting, setSubmitting] = useState(false);
+  const [error, setError] = useState('');
+  const newdle = useSelector(getNewdle);
+  const finalDate = useSelector(getFinalDate);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    return () => {
+      dispatch(clearFinalDate());
+    };
+  }, [dispatch]);
+
+  async function update() {
+    setError('');
+    setSubmitting(true);
+    let updatedNewdle;
+    try {
+      updatedNewdle = await client.updateNewdle(newdle.code, {final_dt: finalDate});
+    } catch (exc) {
+      setSubmitting(false);
+      setError(exc.toString());
+      return;
+    }
+    dispatch(updateNewdle(updatedNewdle));
+    setSubmitting(false);
+  }
+
+  return (
+    <Container text>
+      {newdle && !newdle.final_dt && (
```

Comment:
I'd bail out early with a Loader until `newdle` exists instead of checking it here and below.